src/parser.rs

```diff
@@ -125,6 +141,32 @@ pub fn inline<T>(result: Result<T, ParserError>) -> Result<T, (ParserError, Toke
 	result.map_err(|error| (error, Token::Terminator))
 }
 
+pub fn animation(lexer: &mut Lexer, check_with: bool) -> Result<Option<AnimationDeclaration>, (ParserError, Token)> {
+	if check_with {
+		match inline(lexer.token())? {
+			None | Some(Token::Terminator) => return Ok(None),
+			Some(Token::Identifier(string)) if string == "with" => {}
+			Some(token) => return Err((ParserError::UnexpectedToken, token)),
```

Comment:
The error return is not supposed to specify the token that was given but instead of what token to skip to. For example, if an error occurs inside of a single line statement, you would return the `Token::Terminator` variant to tell the parser to skip and consume a terminator token. This is used for error recovery.

Reply:
Done